src/server/scripts/Pet/pet_dk.cpp

```diff
@@ -105,6 +119,109 @@ class npc_pet_dk_ebon_gargoyle : public CreatureScript
                 // Despawn as soon as possible
                 me->DespawnOrUnsummon(Seconds(4));
             }
+
+            void UpdateAI(uint32 diff) override
+            {
+                events.Update(diff);
+
+                if (despawnTimer > 0)
+                {
+                    if (despawnTimer > diff)
+                        despawnTimer -= diff;
+                    else
+                        me->DespawnOrUnsummon();
+
+                    return;
+                }
+
+                if (spellCastTime <= diff)
```

Comment:
why not handling it with event map/taskscheduler?

Reply:
Okey i move event.